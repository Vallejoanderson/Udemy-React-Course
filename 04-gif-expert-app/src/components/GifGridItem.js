import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( { title, url } ) => {

    return (
        <div className="card animate__animated animate__zoomIn">
            <img src={ url } alt={ title }/>
            <p> { title }</p>
        </div>
    )
}

GifGridItem.propTypes = { 
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
 }

/*  Tarea
    1. Enzyme
    2. Enzyme to Json
    3. Debe de mostrar el componente correctamente
        * shallow
        * wrapper
        * wrapper .toMatchSsnapshot()
*/