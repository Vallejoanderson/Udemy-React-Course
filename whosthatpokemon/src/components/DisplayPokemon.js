

export const DisplayPokemon = async({ pokemones }) => {
    // const randomNumber = Math.floor(Math.random() * 3);
    // const image = await pokemones[randomNumber].image;
    // console.log(pokemones);
    return (
        <img src="" alt="Random pokemon" />
    );
}