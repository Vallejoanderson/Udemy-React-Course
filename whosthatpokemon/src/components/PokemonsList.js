

export const PokemonsList = ( { pokemons } ) => {
    const checkChoice = () => {
        
    }

    return(
        <ol>
            {
                pokemons.map( pokemon => 
                    <button onClick={ () => { console.log( pokemon.name ) } }>{ pokemon.name }</button>)
            }
        </ol>
    );
}