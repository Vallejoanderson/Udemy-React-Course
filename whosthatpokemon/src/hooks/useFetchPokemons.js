import { useState, useEffect } from 'react'
import getPokemon from '../helpers/getPokemon.js'

export const useFetchPokemons = () => {
    const [ pokemons, setPokemons ] = useState([]);    
    
    useEffect( () => {
        for( let i = 0; i < 4; i++)
        {
            getPokemon()
                .then( newPokemon => setPokemons( all => [ 
                    { name: newPokemon.name, image: newPokemon.img }, ...all]) )
        }
    }, []
    )

    return pokemons;

}