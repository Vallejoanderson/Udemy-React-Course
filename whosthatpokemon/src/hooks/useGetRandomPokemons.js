import { useState } from 'react'
import getPokemon from '../helpers/getPokemon.js'

export const useGetRandomPokemons = () => {

    const arr = [];
    const [ pokemons, setPokemons ] = useState([]);
    for( let i = 0; i < 4; i++)
    {
        getPokemon()
             .then( newPokemon => 
                arr = [{ name: newPokemon.name, image: newPokemon.img }, ...arr ] ) 
    }

    setPokemons( [...arr] );

    return pokemons;

}