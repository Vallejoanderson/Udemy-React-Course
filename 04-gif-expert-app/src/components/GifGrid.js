import React, { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ( { category } ) => {

    const [images, setImages] = useState([])
    
    useEffect( () => {
        getGifs( category )
            .then( imgs => setImages( imgs) ) 
    }, [ category ] )

    
    //getGifs();

    return (
        <>
            <h3>{  category }</h3>
            <div className="card-grid">
                {
                        images.map( img => //Desestructuraci√≥n de objecto images
                            <GifGridItem 
                                key={img.id}
                                { ...img }
                            />
                        )
                }
            </div>
        </>
    )
}
