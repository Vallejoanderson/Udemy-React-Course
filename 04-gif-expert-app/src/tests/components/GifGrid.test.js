import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs')


describe('Pruebas en GifGrid', () => {

    const category = 'Soyunacategoria'
    
    test('Obtener snapshot inicial de <GifGrid />', () => {

        useFetchGifs.mockReturnValue({

            data: [],
            loading: true,

        });

        const wrapper = shallow( <GifGrid key = { category } category = { category } /> );
        expect( wrapper ).toMatchSnapshot();

    })

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cual/quierocosa,jpg',
            title: 'Cualquier cosa',
        }]

        useFetchGifs.mockReturnValue({

            data: gifs,
            loading: false,

        });

        const wrapper = shallow( <GifGrid key = { category } category = { category } /> );
        
        expect(wrapper ).toMatchSnapshot();

    })

})
