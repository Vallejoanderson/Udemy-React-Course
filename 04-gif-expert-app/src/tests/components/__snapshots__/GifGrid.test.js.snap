// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    Soyunacategoria
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cual/quierocosa,jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid Obtener snapshot inicial de <GifGrid /> 1`] = `
<Fragment>
  <h3>
    Soyunacategoria
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
